sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/CosmosDb.java

```diff
@@ -70,14 +70,12 @@ public static String urlEncode(String url, String enc) {
         }
     }
 
-    public static Map<String, String> generateHeaders(final String partition, final String dbToken) {
-        Map<String, String> headers = new HashMap<String, String>() {{
-            put("x-ms-documentdb-partitionkey", String.format("[\"%s\"]", partition));
-            put("x-ms-version", "2018-06-18");
-            put("x-ms-date", nowAsRFC1123());
-            put("Content-Type", "application/json");
-            put("Authorization", urlEncode(dbToken));
-        }};
+    public static Map<String, String> addRequiredHeaders(final Map<String, String> headers, final String partition, final String dbToken) {
```

Comment:
Code convetion: Local variables or parameters take final only when needed.
Static or member variables however must be final when they can.